sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/ShowTablesSchema.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.plans
+
+import org.apache.spark.sql.catalyst.expressions.AttributeReference
+import org.apache.spark.sql.types.{BooleanType, StringType, StructType}
+
+private[sql] object ShowTablesSchema {
+  val schema = StructType.fromAttributes(attributes())
+
+  def attributes(): Seq[AttributeReference] = Seq(
```

Comment:
Attributes shouldn't be reused between plans. These should be moved into `ShowTables` so that the attribute ID is assigned each time it is instantiated.

Reply:
This is `def` so new attribute ID will be created whenever `attributes` is invoked. Am I missing something?

Reply:
No, you're right. I missed that it was a def.